azurerm/resource_arm_container_group.go

```diff
@@ -323,6 +357,29 @@ func resourceArmContainerGroupDelete(d *schema.ResourceData, meta interface{}) e
 	return nil
 }
 
+func flattenContainerGroupImageRegistryCredentials(d *schema.ResourceData, credentials *[]containerinstance.ImageRegistryCredential) *[]interface{} {
+	ircConfigs := make([]interface{}, 0, len(*credentials))
+	ircConfigsOld := d.Get("image_registry_credential").([]interface{})
+
+	for _, irc := range *credentials {
+		ircConfig := make(map[string]interface{})
+		ircConfig["server"] = *irc.Server
+		ircConfig["username"] = *irc.Username
+		// search for password in the old configs
+		for _, oldIrcConfig := range ircConfigsOld {
```

Comment:
I'm not entirely sure, but i think the index will always be the same (otherwise each plan/apply we would attempt to update) so you might beable to simplify this to:

```go
for i, oldIrcConfig := range ircConfigsOld {
...

    if v, ok := d.GetOk(fmt.Sprintf("image_registry_credential.%d.password", i)); ok {
        block["password"] = v.(string)
    }
}
```